### PR TITLE
[receiver/mongodb]: Add uptime/health metrics

### PR DESCRIPTION
**Pull request by davidji99**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17021

### CODE REVIEW

**Comment by djaglowski** [comment]
I am not sure we should add health / uptime metrics until https://github.com/open-telemetry/oteps/pull/185 is decided.

**Comment by davidji99** [comment]
> I am not sure we should add health / uptime metrics until [open-telemetry/oteps#185](https://github.com/open-telemetry/oteps/pull/185) is decided.

Thanks for the referring me to doc. The proposal's mention of Prometheus hits home as I/my team often uses the Prometheus scrapers as a guide to what needs to be contributed to its OTel counterpart. We do this comparison because our team's customers have been using Prometheus and/or Datadog for monitoring ES, mongodb, etc.

Now do you know if there is any traction with the proposal? All of the comments are over one year old and I do see the health/uptime metric in Elasticsearchreceiver. I'm happy to make changes to the PR as needed but I'd like to know if the PR will be accepted in any form. THanks.

**Comment by djaglowski** [comment]
> Now do you know if there is any traction with the proposal? All of the comments are over one year old and I do see the health/uptime metric in Elasticsearchreceiver. I'm happy to make changes to the PR as needed but I'd like to know if the PR will be accepted in any form. Thanks.

I have not followed this closely, but it has been discussed more recently [here](https://github.com/open-telemetry/opentelemetry-specification/issues/2923#issuecomment-1315581240). @jmacd, any chance you can advise on the status of this, or whether/how to move forward here?

**Comment by davidji99** [comment]
> > Now do you know if there is any traction with the proposal? All of the comments are over one year old and I do see the health/uptime metric in Elasticsearchreceiver. I'm happy to make changes to the PR as needed but I'd like to know if the PR will be accepted in any form. Thanks.
> 
> I have not followed this closely, but it has been discussed more recently [here](https://github.com/open-telemetry/opentelemetry-specification/issues/2923#issuecomment-1315581240). @jmacd, any chance you can advise on the status of this, or whether/how to move forward here?

@jmacd Pinging you for a comment.

**Comment by davidji99** [comment]
@djaglowski Is this PR essentially going to be frozen until a decision will be made regarding health/uptime? I'd like to know to determine whether or not to fork this receiver for our collector distribution in the interim.

File: receiver/mongodbreceiver/metadata.yaml [hunk 1]
```diff
@@ -324,3 +324,22 @@ metrics:
       aggregation: cumulative
       monotonic: true
     attributes: [database, lock_type, lock_mode]
+  mongodb.health:
+    enabled: false
+    description: The health status of mongodb.
```
Comment (djaglowski) [inline receiver/mongodbreceiver/metadata.yaml:329]:
```suggestion
    description: The health status of the server.
```

Or is this per node? Either way, let's be as specific as possible here.

**Review by djaglowski — COMMENTED** [review]
Since we don't have a standard yet, I think best effort is appropriate here. This looks pretty good to me, just a couple points of feedback.

File: receiver/mongodbreceiver/metadata.yaml (hunk 1 above)
Comment (davidji99) [reply receiver/mongodbreceiver/metadata.yaml:329]: should be the server

**Comment by davidji99** [comment]
@djaglowski Updated

**Review by djaglowski — APPROVED** [review]

**Comment by datsabk** [comment]
@djaglowski - Doesn't look like health metrics are getting emitted for v5.0. Is there any additional configuration for v5.0?

**Pull request merged**